### PR TITLE
Move parse to worker

### PR DESCRIPTION
**Pull request by anandkumarpatel**
- move parsing logic to worker
- add test event to ensure we get an event
- add 10 second timeout to events
# code review
- [x] @podviaznikov
# test
- [x] runnable_tester

### CODE REVIEW

File: lib/listener.js [hunk 1]
```diff
@@ -36,30 +30,88 @@ Listener.prototype.start = function () {
...
- * Emit stopped event
+ * publish event to rabbit
+ * @param  {Buffer} chuck from docker event stream
  */
-Listener.prototype.stop = function () {
-  log.info('Listener.prototype.stop')
-  status.docker_connected = false
-  this.emit('stopped')
+Listener.prototype.publishEvent = function (chuck) {
+  if (chuck) { return }
```
Comment (podviaznikov) [inline lib/listener.js]: chuck or chunk?

**Comment by podviaznikov** [comment]
one comment, looks good

File: lib/listener.js (hunk 1 above)
Comment (ykumar6) [inline lib/listener.js]: chucky

File: lib/listener.js [hunk 2]
```diff
@@ -36,30 +30,88 @@ Listener.prototype.start = function () {
...
-    log.trace('start: got event stream')
+    log.trace('start - got event stream')
 
     status.docker_connected = true
 
+    // timeout is used to ensure we are getting events
+    var timeout = setTimeout(function () {
+      log.error('timeout getting events')
+      self.handleClose()
+    }, process.env.EVENT_TIMEOUT)
```
Comment (bkendall) [inline lib/listener.js]: so, this will kill the process if we don't get anything, right?

File: lib/listener.js [hunk 3]
```diff
@@ -36,30 +30,88 @@ Listener.prototype.start = function () {
...
     eventStream
-      .on('error', this.handleError.bind(this))
-      .on('close', this.handleClose.bind(this))
-      .on('data', this.publisher.write.bind(this.publisher))
+      .on('error', self.handleError.bind(self))
+      .on('close', self.handleClose.bind(self))
+      .on('data', self.publishEvent.bind(self))
+      .once('data', function () {
+        log.info('start - event data received')
+        clearTimeout(timeout)
```
Comment (bkendall) [inline lib/listener.js:51]: I only see you clearing the timeout here... do you want to re-set this timeout again?

File: lib/listener.js (hunk 2 above)
Comment (ykumar6) [inline lib/listener.js]: We're using a timeout to detect we're not getting events? How is this robust?

Comment (anandkumarpatel) [inline lib/listener.js]: yep, that is the plan

File: lib/listener.js (hunk 3 above)
Comment (anandkumarpatel) [inline lib/listener.js:51]: no the only time we miss events is on connect

**Comment by podviaznikov** [comment]
saw a demo

**Pull request merged**